pkg/skaffold/test/custom/custom_test.go

```diff
@@ -0,0 +1,232 @@
+		tmpDir := t.NewTempDir().Touch("test.yaml")
+
+		custom := latest.CustomTest{
+			Command:        "echo Running Custom Test command.",
+			TimeoutSeconds: 10,
+			Dependencies: &latest.CustomTestDependencies{
+				Command: "echo [\"file1\",\"file2\",\"file3\"]",
+				Paths:   []string{"**"},
+				Ignore:  []string{"b*"},
+			},
+		}
+
+		cfg := &mockConfig{
+			workingDir: tmpDir.Root(),
+			tests: []*latest.TestCase{{
+				ImageName:      "image",
+				StructureTests: []string{"test.yaml"},
+				CustomTests:    []latest.CustomTest{custom},
+			}},
+		}
+
+		testRunner, err := New(cfg, cfg.workingDir, custom)
+		t.CheckNoError(err)
+		err = testRunner.Test(context.Background(), ioutil.Discard, nil)
+
+		t.CheckNoError(err)
+	})
+}
+
+func TestCustomCommandError(t *testing.T) {
+	tests := []struct {
+		description   string
+		custom        latest.CustomTest
+		shouldErr     bool
+		expectedError string
+	}{
+		{
+			description: "Non zero exit",
+			custom: latest.CustomTest{
+				Command: "exit 20",
+			},
+			shouldErr:     true,
+			expectedError: "exit status 20",
+		},
+		{
+			description: "Command timed out",
+			custom: latest.CustomTest{
+				Command:        "sleep 20",
+				TimeoutSeconds: 2,
+			},
+			shouldErr:     true,
+			expectedError: "context deadline exceeded",
+		},
+	}
+	for _, test := range tests {
+		testutil.Run(t, "Testing new custom test runner", func(t *testutil.T) {
+			tmpDir := t.NewTempDir().Touch("test.yaml")
+
+			if runtime.GOOS == Windows {
+				test.expectedError = "exit status"
+			}
+
+			cfg := &mockConfig{
+				workingDir: tmpDir.Root(),
+				tests: []*latest.TestCase{{
+					ImageName:      "image",
+					StructureTests: []string{"test.yaml"},
+					CustomTests:    []latest.CustomTest{test.custom},
+				}},
+			}
+
+			testRunner, err := New(cfg, cfg.workingDir, test.custom)
+			t.CheckNoError(err)
+			err = testRunner.Test(context.Background(), ioutil.Discard, nil)
+
+			t.CheckError(test.shouldErr, err)
+			if test.expectedError != "" {
+				t.CheckErrorContains(test.expectedError, err)
+			}
+		})
+	}
+}
+
+func TestTestDependenciesCommand(t *testing.T) {
+	testutil.Run(t, "Testing new custom test runner", func(t *testutil.T) {
+		tmpDir := t.NewTempDir().Touch("test.yaml")
+
+		custom := latest.CustomTest{
+			Command: "echo Hello!",
+			Dependencies: &latest.CustomTestDependencies{
+				Command: "echo [\"file1\",\"file2\",\"file3\"]",
+			},
+		}
+
+		cfg := &mockConfig{
+			workingDir: tmpDir.Root(),
+			tests: []*latest.TestCase{{
+				ImageName:      "image",
+				StructureTests: []string{"test.yaml"},
+				CustomTests:    []latest.CustomTest{custom},
+			}},
+		}
+
+		if runtime.GOOS == Windows {
+			t.Override(&util.DefaultExecCommand, testutil.CmdRunOut(
+				"cmd.exe /C echo [\"file1\",\"file2\",\"file3\"]",
+				"[\"file1\",\"file2\",\"file3\"]",
+			))
+		} else {
+			t.Override(&util.DefaultExecCommand, testutil.CmdRunOut(
```

Comment:
Use the same apprach for `TestCustomCommandError` and `TestNewCustomTestRunner`

Reply:
There are many special cases in the tests of skaffold projects for the platform specific errors. Here is an example:
https://github.com/GoogleContainerTools/skaffold/blob/ba2390db18ad15ccf1eefbce39d6dd8f34016914/pkg/skaffold/kubernetes/portforward/kubectl_forwarder_test.go#L148

Reply:
please change it.  it is not required in this unit test. The test you pointed actually checking if a command is running in background and has extra logic of terminating the command in unit test.

Reply:
If I don't special case it, the test will fail on the PR. This is because the actual commands run are different on different platforms:
On Windows it is `cmd.exe /C`
On non-windows it is `sh -c`

See the code below:
https://github.com/GoogleContainerTools/skaffold/blob/55659bd416070ead13e9a3fe65519ec57cdceb9e/pkg/skaffold/test/custom/custom.go#L83

Here is another example of specialcasing:
https://github.com/GoogleContainerTools/skaffold/blob/ba2390db18ad15ccf1eefbce39d6dd8f34016914/pkg/skaffold/build/custom/script_test.go#L141

Should I disable it on windows like this?:
https://github.com/GoogleContainerTools/skaffold/blob/ba2390db18ad15ccf1eefbce39d6dd8f34016914/pkg/skaffold/docker/auth_test.go#L54